provisioner/remote_files.go

```diff
@@ -0,0 +1,60 @@
+package provisioner
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"github.com/aws/aws-sdk-go/service/kms"
+	"github.com/aws/aws-sdk-go/service/kms/kmsiface"
+	"gopkg.in/yaml.v2"
+)
+
+func makeArchive(input string, kmsKey string, kmsClient kmsiface.KMSAPI) ([]byte, error) {
+	var data remoteData
+	err := yaml.UnmarshalStrict([]byte(input), &data)
+	if err != nil {
+		return nil, err
+	}
+
+	var filesTar bytes.Buffer
+	gzw := gzip.NewWriter(&filesTar)
+	defer gzw.Close()
+	tarWriter := tar.NewWriter(gzw)
+	defer tarWriter.Close()
+	for _, remoteFile := range data.Files {
+		var compressionInput []byte
+		var finalPath string
+
+		decoded, err := base64Decode(remoteFile.Data)
+		if err != nil {
+			return nil, fmt.Errorf("failed to decode data: %v for file: %s", remoteFile.Data, remoteFile.Path)
```

Comment:
Maybe we should not log the data in case it's supposed to be encrypted :)

Reply:
Fix in [PR](https://github.com/zalando-incubator/cluster-lifecycle-manager/pull/147)